common/src/main/java/com/microsoft/identity/common/internal/result/MsalBrokerResultAdapter.java

```diff
@@ -510,10 +515,22 @@
                 packageName,
                 className
         );
-        intent.putExtras(resultBundle);
+        intent.putExtras(interactiveRequestBundle);
         return intent;
     }
 
+    /**
+     * AccountManager strategy is currently returning result in a different format compared with the other strategies.
+     * We should make sure this does NOT happen going forward.
+     */
+    public Bundle extractInteractiveRequestBundleFromResultBundle(@NonNull final Bundle resultBundle) {
+        final Intent interactiveRequestIntent = resultBundle.getParcelable(AccountManager.KEY_INTENT);
+        if (interactiveRequestIntent != null) {
```

Comment:
Is there a way to be explicit about when we want to have this behavior (ONLY when talking over accountManager) rather than implicit, so that it doesn't become an unintended feature?

Reply:
I actually moved this back from AccountManager Strategy, and the reason is that the ipc strategy itself should not have to care about the content of the payload. It's a bug on our side that sends a different payload for each communication channel.

this layer won't/shouldn't be aware of the ipc strategy either (unless we modify the BrokerOperation interface, which I'm not keen of) - so... the answer is no, I can't think of any clean way to handle this.